XenAdmin/TabPages/DockerDetailsPage.cs

```diff
@@ -41,47 +43,76 @@ namespace XenAdmin.TabPages
 {
     public partial class DockerDetailsPage : BaseTabPage
     {
-        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
         private const int REFRESH_INTERVAL = 20000;
 
-        private IXenObject _xenObject;
-        private DockerContainer _container;
-        private VM _vmResideOn;
-        private Host _hostResideOn;
-        private string _resultCache;
+        private IXenObject xenObject;
+        private VM parentVM;
+        private Host host;
+        private string cachedResult;
 
         public IXenObject XenObject
         {
             get
             {
                 Program.AssertOnEventThread();
-                return _xenObject;
+                return xenObject;
             }
             set
             {
                 Program.AssertOnEventThread();
+                RefreshButton.Enabled = true;
 
                 if (value == null)
                     return;
 
-                if (_xenObject != value)
+                if (xenObject != value)
                 {
-                    _xenObject = value;
-                    if (_xenObject is DockerContainer)
+                    xenObject = value;

```

Comment:
I know this was existing code, but I have just now found the following that would be nice to be fixed.

xenObject will be assigned even if it is not DockerContainer. I think an assertion would be good to have here.